app/jobs/reports/daily_registrations_report.rb

```diff
@@ -0,0 +1,91 @@
+module Reports
+  class DailyRegistrationsReport < BaseReport
+    REPORT_NAME = 'daily-registrations-report'
+
+    attr_reader :report_date
+
+    def perform(report_date)
+      @report_date = report_date
+
+      _latest, path = generate_s3_paths(REPORT_NAME, 'json', now: report_date)
+      body = report_body.to_json
+
+      [
+        bucket_name, # default reporting bucket
+        IdentityConfig.store.s3_public_reports_enabled && public_bucket_name,
+      ].select(&:present?).
+        each do |bucket_name|
+        upload_file_to_s3_bucket(
+          path: path,
+          body: body,
+          content_type: 'application/json',
+          bucket: bucket_name,
+        )
+      end
+    end
+
+    def finish
+      report_date.end_of_day
+    end
+
+    def report_body
+      results = (total_users.to_a + fully_registered_users.to_a).
+        group_by { |row| row['date'] }.
+        map do |date, rows|
+          {
+            date: date,
+            total_users: rows.map { |r| r['total_users'] }.compact.first || 0,
+            fully_registered_users: rows.map { |r| r['fully_registered_users'] }.compact.first || 0,
+          }
+        end.sort_by { |elem| elem[:date] }
+
+      {
+        finish: finish,
+        results: results.as_json,
+      }
```

Comment:
one note: other reports collect a single day at a time (to keep calculations relatively cheap), and the frontend will make one request per day it's querying.

However, for this report, we always want to know all of time. So rather than have clients make requests going back each day to 2017 (1000+ requests), I opted to save client time so the clients only need to query one "day" to get all time.

I think it's a worthwhile tradeoff, but open to feedback!

Reply:
Do we want to write over the same file in S3 then? It could get quite large.

Reply:
no this will make a new file every day, just like the current reports. A report from last week is about 152kb, so not that big IMO